File: Python/bubbleSort.py
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(n-1):  # Always iterates fully, even if sorted
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]  # Swap elements
    return arr

arr = [5, 2, 9, 1, 5, 6]
print("Unsorted:", arr)
print("Sorted:", bubble_sort(arr))  # O(n^2) complexity

File: JavaScript/bubbleSort.js
function bubbleSort(arr) {
    let n = arr.length;
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n - 1; j++) { // Always loops fully, even if sorted
            if (arr[j] > arr[j + 1]) {
                let temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    return arr;
}

let numbers = [5, 3, 8, 4, 2];
console.log(bubbleSort(numbers));
